pkg/odo/cli/component/component.go

```diff
@@ -1,104 +1,46 @@
 package component
 
 import (
-	"fmt"
-
-	"github.com/redhat-developer/odo/pkg/odo/cli/application"
-	"github.com/redhat-developer/odo/pkg/odo/cli/project"
-
-	"github.com/redhat-developer/odo/pkg/log"
 	"github.com/redhat-developer/odo/pkg/odo/util/completion"
 
-	"github.com/golang/glog"
-	"github.com/redhat-developer/odo/pkg/component"
 	"github.com/redhat-developer/odo/pkg/odo/genericclioptions"
 	odoutil "github.com/redhat-developer/odo/pkg/odo/util"
 	"github.com/spf13/cobra"
 )
 
-var componentShortFlag bool
+// RecommendedComponentCommandName is the recommended component command name
+const RecommendedComponentCommandName = "component"
 
-// componentCmd represents the component command
-var componentCmd = &cobra.Command{
-	Use:   "component",
-	Short: "Components of application.",
-	Example: fmt.Sprintf("%s\n%s",
-		componentGetCmd.Example,
-		componentSetCmd.Example),
-	// 'odo component' is the same as 'odo component get'
-	// 'odo component <component_name>' is the same as 'odo component set <component_name>'
-	Run: func(cmd *cobra.Command, args []string) {
-		if len(args) > 0 && args[0] != "get" && args[0] != "set" {
-			componentSetCmd.Run(cmd, args)
-		} else {
-			componentGetCmd.Run(cmd, args)
-		}
-	},
-}
+// NewCmdComponent implements the component odo command
+func NewCmdComponent(name, fullName string) *cobra.Command {
 
-var componentGetCmd = &cobra.Command{
-	Use:   "get",
-	Short: "Get currently active component",
-	Long:  "Get currently active component.",
-	Example: `  # Get the currently active component
-  odo component get
-	`,
-	Args: cobra.NoArgs,
-	Run: func(cmd *cobra.Command, args []string) {
-		glog.V(4).Infof("component get called")
-		context := genericclioptions.NewContext(cmd)
-		component := context.ComponentAllowingEmpty(true)
+	componentGetCmd := NewCmdGet(RecommendedGetCommandName, odoutil.GetFullName(fullName, RecommendedGetCommandName))
+	componentSetCmd := NewCmdSet(RecommendedSetCommandName, odoutil.GetFullName(fullName, RecommendedSetCommandName))
 
-		if componentShortFlag {
-			fmt.Print(component)
-		} else {
-			if component == "" {
-				log.Error("No component is set as current")
-				return
+	// componentCmd represents the component command
+	var componentCmd = &cobra.Command{
+		Use:   name,
+		Short: "Components of application.",
+		/*Example: fmt.Sprintf("%s\n%s",
```

Comment:
Can we please get rid of these comments?

Reply:
Sure